sdk/monitor/monitor-query/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,389 @@
+  - [Troubleshooting unsupported granularity](#troubleshooting-unsupported-granularity-for-metrics-query)
+
+## General troubleshooting
+
+### Enable client logging
+
+To troubleshoot issues with Azure Monitor Query library, it's important to first enable logging to monitor the behavior of the application. The errors and warnings in the logs generally provide useful insights into what went wrong and sometimes include corrective actions to fix issues.
+
+The Azure client libraries for JavaScript allow you to enable logging either through the environment variable or at runtime.
+
+The following log levels are supported. They're listed from most to least verbose.
+
+- verbose
+- info
+- warning
+- error
+
+When setting a log level, either programmatically or via the `AZURE_LOG_LEVEL` environment variable, any logs that are written using a log level equal to or less than the one you choose will be emitted. For example, setting the log level to `warning` will cause all logs that have the log level `warning` or `error` to be emitted.
+
+#### Logging via environment variable
+
+To see a log of HTTP requests and responses, set the `AZURE_LOG_LEVEL` environment variable to `info`.
+
+Add the following to the .env
+```text
+AZURE_LOG_LEVEL = info
+```
+
+Add the following to the code 
+```ts
+import * as dotenv from "dotenv";
+dotenv.config({ path: ".env" });
+```
+
+#### Logging using setLogLevel
+
+Alternatively, logging can be enabled at runtime by calling `setLogLevel` in the `@azure/logger`:
+
+```ts
+import { setLogLevel } from "@azure/logger";
+
+setLogLevel("info");
+```
+
+**NOTE**: When logging the body of request and response, ensure that they don't contain confidential information.
+
+For detailed instructions on how to enable logs, see the [@azure/logger package docs](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/core/logger).
+
+### Authentication errors
+
+Azure Monitor Query supports Azure Active Directory authentication. Both `logsQueryClient` and `metricsQueryClient` have methods to set the `credential`. To provide a valid credential, you can use the `@azure/identity` dependency. For more details on getting started, refer to the [README](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/monitor/monitor-query/README.md#create-the-client) of Azure Monitor Query library. You can also refer to the [Azure Identity documentation](https://aka.ms/azsdk/js/identity/troubleshoot) for more details on the various credential types supported in `@azure/identity`.
+
+## Troubleshooting logs query
+
+### Troubleshooting insufficient access error for logs query
+
+If you get an HTTP error with status code 403 (Forbidden), it means the provided credentials lack sufficient permissions to query the workspace.
+
+```json
+ {
+  "name": "RestError",
+  "code": "InsufficientAccessError",
+  "statusCode": 403,
+  "request": {
+    "url": "https://api.loganalytics.io/v1/workspaces/08e1f3a5-9cf2-47a4-94b6-285bf25dc4be/query",
```

Comment:
do we have to worry about the unique id of the workspace being public here?

Reply:
Good point. Let's use a placeholder like `<workspace_id>` instead of the actual value.

Reply:
updated